pkg/machine/shim/host.go

```diff
@@ -447,6 +449,19 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDe
 		return fmt.Errorf("reload config: %w", err)
 	}
 
+	// if the machine cannot continue starting due to a signal, ensure the state
+	// reflects the machine is no longer starting
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		for _ = range signalChan {
+			mc.Starting = false
+			if err := mc.Write(); err != nil {
+				logrus.Error(err)
+			}
+		}
+	}()
```

Comment:
There is nothing cancelling this function so this would run forever!

Not the biggest deal in terms of podman machine start but the actual issue if the signal comes in after we successfully started you always unconditionally overwrite which is likely not cause real issues right now but if the API ever changes or Start() is called in a long running process I could see that going wrong badly.

This likely need some context added which can be cancelled and then you have a select inside a for loop.

Reply:
Oh and worse you need to os.Exit() or somthing here. Otherwise you now ignored the signal and the code continues to run

Reply:
If we use os.Exit() then actually solves the runs forever problem logically so no need to over complicate it then I guess.

Reply:
Should be fixed now. If the channel receives a signal, it will call `os.Exit(1)`. However, if the machine transitions into the running state and the channel  hasn't received a signal, I will close the signal which should also stop the goroutine from running. Let me know what you think